src/game/spawn/mod.rs

```diff
@@ -1,12 +1,30 @@
-//! Handles spawning of entities. Here, we are using
+//! Handles spawning of well defined game entities. Here, we are using
 //! [observers](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Observer.html)
-//! for this, but you could also use `Events<E>` or `Commands`.
+//! to separate spawn requests from spawn resource requirements, like assets.
 
-use bevy::prelude::*;
+use bevy::{ecs::system::EntityCommands, prelude::*};
 
 pub(crate) mod level;
 pub(crate) mod player;
 
 pub(super) fn plugin(app: &mut App) {
     app.add_plugins((level::plugin, player::plugin));
 }
+
+trait SpawnTriggerHelper {
+    /// Returns commands for the targeted entity if one exists.
+    /// Spawns a new entity otherwise.
+    fn get_trigger_entity_or_spawn<E>(&mut self, trigger: Trigger<E>) -> EntityCommands;
+}
+
+impl<'a, 'b> SpawnTriggerHelper for Commands<'a, 'b> {
+    fn get_trigger_entity_or_spawn<E>(&mut self, trigger: Trigger<E>) -> EntityCommands {
+        let entity = trigger.entity();
+        // Cannot use `match` due to compile time trait restraints
+        if entity == Entity::PLACEHOLDER {
+            self.spawn_empty()
+        } else {
+            self.entity(entity)
+        }
+    }
+}
```

Comment:
This can be deleted if we expect these events to always be triggered with an entity. I think that's valid, since we're providing `spawn_trigger` as the interface for triggering these events.

Then you can do `commands.entity(trigger.entity()).insert(...)` etc

Reply:
Spawn trigger is fully separate from this api, its not obvious